core/chains/terra/terratxm/txm.go

```diff
@@ -0,0 +1,355 @@
+package terratxm
+
+import (
+	"encoding/hex"
+	"strings"
+	"time"
+
+	"github.com/smartcontractkit/chainlink/core/services/keystore/keys/terrakey"
+
+	"github.com/tendermint/tendermint/crypto/tmhash"
+
+	"github.com/pkg/errors"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	txtypes "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/smartcontractkit/chainlink/core/services/keystore"
+	wasmtypes "github.com/terra-money/core/x/wasm/types"
+
+	terraclient "github.com/smartcontractkit/chainlink-terra/pkg/terra/client"
+	"github.com/smartcontractkit/chainlink/core/logger"
+	"github.com/smartcontractkit/chainlink/core/services"
+	"github.com/smartcontractkit/chainlink/core/services/pg"
+	"github.com/smartcontractkit/chainlink/core/utils"
+	"github.com/smartcontractkit/sqlx"
+)
+
+var (
+	_ services.Service = (*Txm)(nil)
+)
+
+const (
+	// MaxMsgsPerBatch The max gas limit per block is 1_000_000_000
+	// https://github.com/terra-money/core/blob/d6037b9a12c8bf6b09fe861c8ad93456aac5eebb/app/legacy/migrate.go#L69.
+	// The max msg size is 10KB https://github.com/terra-money/core/blob/d6037b9a12c8bf6b09fe861c8ad93456aac5eebb/x/wasm/types/params.go#L15.
+	// Our msgs are only OCR reports for now, which will not exceed that size.
+	// There appears to be no gas limit per tx, only per block, so theoretically
+	// we could include 1000 msgs which use up to 1M gas.
+	// To be conservative and since the number of messages we'd
+	// have in a batch on average roughly correponds to the number of terra ocr jobs we're running (do not expect more than 100),
+	// we can set a max msgs per batch of 100.
+	MaxMsgsPerBatch = 100
+
+	// BlocksUntilTxTimeout ~8s per block, so ~80s until we give up on the tx getting confirmed
+	// Anecdotally it appears anything more than 4 blocks would be an extremely long wait.
+	BlocksUntilTxTimeout = 10
+)
+
+// Txm manages transactions for the terra blockchain.
+type Txm struct {
+	starter            utils.StartStopOnce
+	eb                 pg.EventBroadcaster
+	sub                pg.Subscription
+	ticker             *time.Ticker
+	orm                *ORM
+	lggr               logger.Logger
+	tc                 terraclient.ReaderWriter
+	ks                 keystore.Terra
+	stop, done         chan struct{}
+	confirmPollPeriod  time.Duration
+	confirmMaxPolls    int
+	fallbackGasPrice   sdk.DecCoin
+	gasLimitMultiplier float64
+}
+
+// NewTxm creates a txm
+func NewTxm(db *sqlx.DB, tc terraclient.ReaderWriter, fallbackGasPrice string, gasLimitMultiplier float64, ks keystore.Terra, lggr logger.Logger, cfg pg.LogConfig, eb pg.EventBroadcaster, pollPeriod time.Duration) (*Txm, error) {
+	ticker := time.NewTicker(pollPeriod)
```

Comment:
Pls use `utils.WithJitter` here, just to head off any potential thundering herd problem.